routemaster/models/event.rb

```diff
@@ -12,24 +13,26 @@ class Event < Message
 
       VALID_TYPES = %w(create update delete noop)
 
-      attr_reader :topic, :type, :url
+      attr_reader :topic, :type, :url, :data
 
       def initialize(**options)
         super
-        _assert VALID_TYPES.include?(options[:type]), 'bad event type'
+        @type      = options.fetch(:type, nil)
+        @url       = CallbackURL.new options.fetch(:url, nil)
```

Comment:
Ah OK. Maybe using `options[:url]` rather than `fetch` with `nil` would be slightly clearer?